openfl/component/collaborator/collaborator.py

```diff
@@ -252,6 +264,11 @@ def do_task(self, task, round_number):
             func = getattr(self.task_runner, func_name)
             self.logger.debug('Using TaskRunner subclassing API')
 
+        # MICAH CHANGE: pop dynamic args and add to kwargs
+        for key in dynamicarg_tensor_keys:
+            arg_name = arg_name_from_dynamic_task_arg_tensor_key(key)
+            kwargs[arg_name] = input_tensor_dict.pop(key.tensor_name)[0]
+
```

Comment:
Makes sense, but I wonder whether this should trigger an exception or a warning?

Reply:
Do you think this can be just documented as "args defined in dynamic args always override args in task config of the same name" ?

We can have a check in the plan object to raise an error if it finds duplicate args, but I think this approach wouldn't cover other cases (e.g., I see in the aggregator code there is an if statement checking if the taskrunner has a `TASK_REGISTRY` attribute, and if yes it neglects the task config defined in the plan and does some other logc....)

Reply:
for now just add a comment here to document it later

Reply:
I like the documentation approach. The OpenFL team may well change how this works anyhow.